### PR TITLE
Add brainfuck cheatsheet

### PR DESCRIPTION
**Pull request by AdeshAtole**
IA Page: https://duck.co/ia/view/brainfuck_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@AdeshAtole Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=brainfuck cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/brainfuck_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/brainfuck.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+	"id": "brainfuck_cheat_sheet",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/brainfuck.json]: Please convert all your tabs to spaces. We use 4 spaces per indent.

File: share/goodie/cheat_sheets/json/brainfuck.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+				"val": "Increment byte at data pointer"
+			}, {
+				"key": "-",
+				"val": "Decrement byte at data pointer"
+			}, {
+				"key": ".",
+				"val": "Output byte at data pointer"
+			}, {
+				"key": ",",
+				"val": "Accept a byte at input, and store at data pointer"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/brainfuck.json]: I think that comma can go.

File: share/goodie/cheat_sheets/json/brainfuck.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
+    "id": "brainfuck_cheat_sheet",
+    "name": "Brainfuck",
+
+    "metadata": {
+        "sourceName": "Wikipedia",
+        "sourceUrl": "https://en.wikipedia.org/wiki/Brainfuck#Commands"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/brainfuck.json]: I think you should look at some other sources such as [this](http://esolangs.org/wiki/brainfuck) and [this](http://www.muppetlabs.com/~breadbox/bf/). You can copy the definitions word-for-word - you don't have to come up with your own phrasing.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=brainfuck cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/brainfuck_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/brainfuck.json (hunk 2 above)
Comment (AdeshAtole) [inline share/goodie/cheat_sheets/json/brainfuck.json]: Which comma?

File: share/goodie/cheat_sheets/json/brainfuck.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                "val": "Increment the byte at the pointer"
+            }, {
+                "key": "-",
+                "val": "Decrement the byte at the pointer."
+            }, {
+                "key": ".",
+                "val": "Output the byte at the pointer."
+            }, {
+                "key": ",",
+                "val": "Input a byte and store it in the byte at the pointer."
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/brainfuck.json]: Maybe get rid of the periods at the ends of the sentences.

**Comment by GuiltyDolphin** [comment]
Okay, :+1: this LGTM @moollaza.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=brainfuck cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/brainfuck_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=brainfuck cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/brainfuck_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@AdeshAtole Thanks for this.

Our policy is to not show any "adult" content. Typically IA's with expletives (like f*ck) have an `is_adult` flag, but we haven't implemented anything for cheat sheets yet because none have had "adult" content.

I need to check with the team about this one.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=brainfuck cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/brainfuck_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=brainfuck cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/brainfuck_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=brainfuck cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/brainfuck_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
We have "brainfuck" whitelisted internally. Currently it doesn't set off our safesearch filter so I will allow this to move ahead.

https://duckduckgo.com/?q=brainfuck&t=ffab&ia=about

**Pull request merged**